In [9]:
''' 
Para ler e escrever dados em Python, utilizamos as seguintes funções: 
- input: lê UMA linha com dado(s) de Entrada do usuário;
- print: imprime um texto de Saída (Output), pulando linha.  
'''

class ContaBancaria:
    # TODO: Inicialize a conta bancária com o nome do titular, saldo 0 e  liste para armazenar as operações realizadas:
    def __init__(self, titular):
      self.titular = titular
      self.saldo = 0
      self.operacoesList = []

    # TODO: Implemente o método para realizar um depósito, adicione o valor ao saldo e registre a operação:
    def depositar(self, valor):
      self.saldo = self.saldo + valor
      self.operacoesList.append("+" + str(valor))
    # TODO: Implemente o método para realizar um saque:
    def sacar(self, valor):
      # TODO: Verifique se há saldo suficiente para o saque
      if self.saldo >= abs(valor):
        # TODO: Subtraia o valor do saldo (valor já é negativo)
        self.saldo = self.saldo + valor
        self.operacoesList.append(str(valor))
      else:
        # TODO: Registre a operação e retorne a  mensagem de saque negado
        self.operacoesList.append(f"Saque não permitido")

    # TODO: Crie o método para exibir o extrato da conta e junte as operações no formato correto:
    def extrato(self):
      print(f"Operações: {", ".join([str(valor) for valor in self.operacoesList])}; Saldo: {self.saldo}")


nome_titular = input().strip()  
conta = ContaBancaria(nome_titular)  

entrada_transacoes = input().strip()
transacoes = [int(valor) for valor in entrada_transacoes.split(",")]  

for valor in transacoes:
    if valor > 0:
        conta.depositar(valor)
    else:
        conta.sacar(valor)

conta.extrato()

Operações: +100, -50, +200, Saque não permitido; Saldo: 250


## Descrição

Implemente um sistema que gerencie várias contas bancárias. Cada conta será representada como uma instância da classe ContaBancaria criada no desafio anterior. O sistema deve permitir que você crie contas para diferentes titulares e liste todas as contas cadastradas ao final da execução.

## Requisitos
O sistema deve permitir:
Criar contas:  Ao criar uma conta, forneça o nome do titular e o saldo inicial no formato "Titular, SaldoInicial".
Listar contas:  Ao digitar o comando especial "FIM", o sistema deverá listar todas as contas cadastradas no formato especificado.

## Entrada

O sistema deve permitir:

- Criação de contas no formato: "Titular, SaldoInicial".
- Um comando especial "FIM" será usado para encerrar o processo de entrada e listar as contas.

## Saída

Liste todas as contas cadastradas no formato: "Titular: X, Saldo: R$ Y"

## Exemplos

A tabela abaixo apresenta exemplos com alguns dados de entrada e suas respectivas saídas esperadas. Certifique-se de testar seu programa com esses exemplos e com outros casos possíveis.

Entradas

João, 500
Maria, 1000
FIM

Ana, 150
Bruno, 250
FIM

Fernando, 50
Gustavo, 75
Helena, 125
FIM

Saidas:

João: R$ 500, Maria: R$ 1000

Ana: R$ 150, Bruno: R$ 250

Fernando: R$ 50, Gustavo: R$ 75, Helena: R$ 125

In [5]:
''' 
Para ler e escrever dados em Python, utilizamos as seguintes funções: 
- input: lê UMA linha com dado(s) de Entrada do usuário;
- print: imprime um texto de Saída (Output), pulando linha.  
'''

class ContaBancaria:
    def __init__(self, titular, saldo):
        self.titular = titular
        self.saldo = saldo


#TODO: Implemente a classe SistemaBancario:
class SistemaBancario:
    # TODO: Inicialize a lista de contas:
    def __init__(self):
        self.listarContas = []
    
    def armazenar_contas(self, conta):
        self.listarContas.append(conta)

    # TODO: Crie uma nova conta e adicione à lista de contas:
    def criar_conta(self, titular, saldo):
        conta = ContaBancaria(titular, saldo)
        self.armazenar_contas(f"{conta.titular}: R$ {conta.saldo}")

    # TODO: Liste todas as contas no formato "Titular: R$ Saldo":
    def listar_contas(self):
        print(f"{", ".join(self.listarContas)}")

#TODO: Crie uma instância de SistemaBancario:
sistema = SistemaBancario()

while True:
    entrada = input().strip()
    if entrada.upper() == "FIM":  
        break
    titular, saldo = entrada.split(", ")
    sistema.criar_conta(titular, int(saldo))

sistema.listar_contas()

João: R$ 500, Maria: R$ 1000
